src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -165,19 +196,33 @@ public void purge() {
     shooterGate.set(-1 * MoPrefs.getShooterGateSetpoint());
   }
 
-  private double getOpenLoopSetpoint(double openLoopSetpoint) {
-    return Utils.map(openLoopSetpoint, -MAX_FREE_SPEED, MAX_FREE_SPEED, -1, 1);
+  /**
+   * 
+   * @param closedLoopSetpoint The PID setpoint, in RPM
+   * @return A number between -1 and 1 to be used for openloop motor control
+   */
+  private double getOpenLoopSetpoint(double closedLoopSetpoint) {
+    return Utils.map(closedLoopSetpoint, -MAX_FREE_SPEED, MAX_FREE_SPEED, -1, 1);
   }
 
   private boolean isFlywheelReady() {
-    return Math.abs(MoPrefs.getShooterPIDSetpoint() - shooterEncoder.getVelocity()) < MoPrefs
-        .getShooterFlywheelTolerance();
+    return Math.abs(setpoint - shooterEncoder.getVelocity()) < MoPrefs.getShooterFlywheelTolerance();
+  }
+
+  private void updatePIDConstants() {
+    shooterPIDRight.setP(kP, 0);
+    shooterPIDRight.setI(kI, 0);
+    shooterPIDRight.setD(kD, 0);
+    shooterPIDRight.setIZone(kIZ, 0);
+    shooterPIDRight.setFF(kFF, 0);
   }
 
   @Override
   public void periodic() {
-    SmartDashboard.putNumber("Flywheel Speed", shooterEncoder.getVelocity());
-    SmartDashboard.putNumber("Flywheel Position", shooterEncoder.getPosition());
-    SmartDashboard.putBoolean("Flywheel ready?", isFlywheelReady());
+    flywheelSpeed.setDouble(shooterEncoder.getVelocity());
+    isFlywheelReady.setBoolean(isFlywheelReady());
+    updatePIDConstants();
+
+    setpoint = shooterSetpoint.getDouble(4500);
```

Comment:
What's going on here?  Why is the setpoint being updated every periodic and there's a hardcoded constant which is repeated multiple times in the code.